src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -1061,6 +1068,11 @@ - (void)_triggerResubscribeWithReason:(NSString *)reason nodeLikelyReachable:(BO
             caseSessionMgr->ReleaseSession(commissioner->GetPeerScopedId(self->_nodeID.unsignedLongLongValue));
         } errorHandler:nil /* not much we can do */];
     }
+
+    // Reattempt subscription after the above ReleaseSession call to avoid churn
```

Comment:
This should probably more clearly explain why it's still "after" even though the ReleaseSession is async?  Presumably because _reattemptSubscriptionNowIfNeededWithReason will also dispatch to the Matter queue?